### DISCUSSION TITLE
"Concerns About IP Address Exposure When Using the 'Direct' SNI Setting in Reality Protocol"

### DISCUSSION THREAD

**Discussion by us254** [XTLS/Xray-core#2600] [Q&A]
**I am considering setting the SNI to go 'direct'** in `my client-side routing` settings for the **Reality protocol** to potentially **speed up connections and reduce latency**. However, I am concerned about the **potential exposure of my real IP address**. In this context, would the IP address that gets exposed be that of my **local client machine** or my **remote server**?

in the client side config, **if my SNI for reality be  exmple.com**
```
{
  "routing": {
    "domainStrategy": "IPIfNonMatch",
    "rules": [
      {
        "type": "field",
        "domain": [
          "example.com"
        ],
        "outboundTag": "direct"
      }
    ]
  }
}
```

**Comment by AmirHosseinCV** [comment]
It would be your remote server's IP. If you want to bypass some websites, it should be configured on the client side.
